### ISSUE TITLE
Haraka crash when loading queue

### ISSUE THREAD

**Issue by gramakri** [haraka/Haraka#2341]
### system info

Haraka 7c4cf785916fcff6c3 on Cloudron

echo "Haraka | $(haraka -v)"; echo " --- | :--- "; echo "Node | $(node -v)"; echo "OS | $(uname -a)"; echo "openssl | $(openssl version)"

```
Haraka | Haraka.js — Version: 2.8.16
 --- | :--- 
Node | v6.11.3
OS | Linux 1ee0f66fe4b0 4.4.0-112-generic #135-Ubuntu SMP Fri Jan 19 11:48:36 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
openssl | OpenSSL 1.0.2g  1 Mar 2016
```

### Expected behavior

Haraka should not crash

### Observed behavior

```
2018-01-03T17:41:33.981Z [INFO] [-] [core] [outbound] Loading the queue...
2018-01-03T17:41:33.996Z [CRIT] [-] [core] SyntaxError: Unexpected end of JSON input
2018-01-03T17:41:33.996Z [CRIT] [-] [core]     at Object.parse (native)
2018-01-03T17:41:33.996Z [CRIT] [-] [core]     at ReadStream.<anonymous> (/app/haraka/outbound/hmail.js:124:34)
2018-01-03T17:41:33.996Z [CRIT] [-] [core]     at emitOne (events.js:96:13)
2018-01-03T17:41:33.996Z [CRIT] [-] [core]     at ReadStream.emit (events.js:188:7)
2018-01-03T17:41:33.996Z [CRIT] [-] [core]     at readableAddChunk (_stream_readable.js:176:18)
2018-01-03T17:41:33.996Z [CRIT] [-] [core]     at ReadStream.Readable.push (_stream_readable.js:134:10)
2018-01-03T17:41:33.996Z [CRIT] [-] [core]     at onread (fs.js:2018:12)
2018-01-03T17:41:33.996Z [CRIT] [-] [core]     at FSReqWrap.wrapper [as oncomplete] (fs.js:683:17)
2018-01-03T17:41:33.996Z [NOTICE] [-] [core] Shutting down
```

### Steps to reproduce

This is hard to reproduce but it has happened on many of our servers. It appears this happens because the outbound queue is corrupt and thus the JSON.parse fails.

I will send a PR so that atleast haraka doesn't crash but atleast the core issue remains as to why the outbound became corrupt

**Comment by baudehlo** [comment]
There has been known problems with this previously that we think are fixed
in 2.8.17 (soon to be released).

**Comment by msimerson** [comment]
Yup, fixed in #2231. `npm install haraka/Haraka` to install from GitHub for now.
